coordinator/src/main/scala/filodb.coordinator/queryplanner/PlannerHelper.scala

```diff
@@ -293,12 +303,22 @@ trait  PlannerHelper {
   }
 
   def materializeTopLevelSubquery(qContext: QueryContext, tlsq: TopLevelSubquery): PlanResult = {
+    val qp = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+    val newQp = qp.copy(
+      promQl = LogicalPlanParser.convertToQuery(tlsq.innerPeriodicSeries),
+      startSecs = tlsq.innerPeriodicSeries.startMs/1000,
+      stepSecs = tlsq.innerPeriodicSeries.stepMs/1000,
+      endSecs = tlsq.innerPeriodicSeries.endMs/1000
+    )
+    val updatedQueryContext = qContext.copy(
+      origQueryParams = newQp
+    )
```

Comment:
We should ideally not update query context. It should be the query that we received at the HTTP entry point.

Reply:
My suggestion would be to leave out the over-time range functions which come into play when we split a namespace across partitions by time. This should be scoped into a different effort.